x-pack/plugins/fleet/server/services/epm/registry/index.ts

```diff
@@ -250,35 +250,59 @@ export async function getInfo(name: string, version: string) {
   });
 }
 
+// Check that the packageInfo exists in cache
+// If not, retrieve it from the archive
+async function getPackageInfoFromArchiveOrCache(
+  name: string,
+  version: string,
+  archiveBuffer: Buffer,
+  archivePath: string
+): Promise<ArchivePackage> {
+  const cachedInfo = getPackageInfo({ name, version });
+
+  if (!cachedInfo) {
+    const { packageInfo } = await generatePackageInfoFromArchiveBuffer(
+      archiveBuffer,
+      ensureContentType(archivePath)
+    );
+    setPackageInfo({ packageInfo, name, version });
+    return packageInfo;
+  } else {
+    return cachedInfo;
+  }
+}
+
 export async function getRegistryPackage(
   name: string,
   version: string,
-  options?: { ignoreUnverified?: boolean; getPkgInfoFromArchive?: boolean }
+  options?: { ignoreUnverified?: boolean }
 ): Promise<{
   paths: string[];
-  packageInfo: RegistryPackage;
+  packageInfo: ArchivePackage;
   verificationResult?: PackageVerificationResult;
 }> {
   const verifyPackage = appContextService.getExperimentalFeatures().packageVerification;
   let paths = getArchiveFilelist({ name, version });
   let verificationResult = verifyPackage ? getVerificationResult({ name, version }) : undefined;
+
+  const {
+    archiveBuffer,
+    archivePath,
+    verificationResult: latestVerificationResult,
+  } = await withPackageSpan('Fetch package archive from archive buffer', () =>
+    fetchArchiveBuffer({
+      pkgName: name,
+      pkgVersion: version,
+      shouldVerify: verifyPackage,
+      ignoreUnverified: options?.ignoreUnverified,
+    })
+  );
+
+  if (latestVerificationResult) {
+    verificationResult = latestVerificationResult;
+    setVerificationResult({ name, version }, latestVerificationResult);
+  }
   if (!paths || paths.length === 0) {
-    const {
-      archiveBuffer,
-      archivePath,
-      verificationResult: latestVerificationResult,
-    } = await withPackageSpan('Fetch package archive from registry', () =>
-      fetchArchiveBuffer({
```

Comment:
This function internally calls `getInfo` to retrieve the `ArchiveBuffer`:
https://github.com/elastic/kibana/blob/4019d378f6e10884c699ba0e11035067b86eb9be/x-pack/plugins/fleet/server/services/epm/registry/index.ts#L312-L330

I wonder if I should get the `ArchiveBuffer` from the cache instead and remove the calls to `getInfo` altogether.